Firestore/core/test/unit/bundle/bundle_serializer_test.cc

```diff
@@ -72,51 +245,825 @@ TEST(BundleSerializerTest, DecodesBundleMetadata) {
   EXPECT_EQ(proto_metadata.total_documents(), actual.total_documents());
 }
 
-TEST(BundleSerializerTest, DecodesInvalidBundleMetadataReportsError) {
+TEST_F(BundleSerializerTest, DecodesInvalidBundleMetadataReportsError) {
   auto proto_metadata = TestBundleMetadata();
 
   std::string json_string;
   MessageToJsonString(proto_metadata, &json_string);
   auto invalid = "123" + json_string;
 
-  BundleSerializer serializer;
-  ReadContext context;
-  serializer.DecodeBundleMetadata(context, invalid);
+  JsonReader reader;
+  bundle_serializer.DecodeBundleMetadata(reader, invalid);
 
-  EXPECT_FALSE(context.ok());
+  EXPECT_FALSE(reader.ok());
 
   // Replace total_bytes to a string unparseable to integer.
   std::string json_copy =
       ReplacedCopy(json_string, "123456789987654321", "xxxyyyzzz");
-  context = ReadContext{};
-  EXPECT_TRUE(context.ok());
-  serializer.DecodeBundleMetadata(context, json_copy);
+  reader = JsonReader{};
+  EXPECT_TRUE(reader.ok());
```

Comment:
Consider breaking this base case assertion out into its own sub-case. At first I thought this was related to parsing the json_copy and was an error.

Reply:
Don't think it's still needed with separate blocks.